src/parser/warlock/demonology/modules/features/SummonDemonicTyrant.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+
+import Analyzer , {SELECTED_PLAYER} from 'parser/core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import Events from 'parser/core/Events';
+
+import StatisticBox from 'interface/others/StatisticBox';
+import SpellIcon from 'common/SpellIcon';
+
+import DemoPets from '../pets/DemoPets';
+
+class SummonDemonicTyrant extends Analyzer {
+  static dependencies = {
+    demoPets: DemoPets,
+  };
+
+  _petsPerCast = [];
+
+
+  constructor(...args){
+    super(...args);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.SUMMON_DEMONIC_TYRANT), this.summonDemonicTyrantCast);
+  }
+
+  summonDemonicTyrantCast(event) {
+    const pets = this.demoPets.currentPets;
+    const countsPerCast = {};
+
+    pets.forEach(pet => {
+      countsPerCast[pet.summonedBy] = (countsPerCast[pet.summonedBy] || 0) + 1;
+    });
+
+    this._petsPerCast.push(countsPerCast);
+  }
+
+
+  statistic() {
+    console.log(this._petsPerCast);
+    const avgPets = (this._petsPerCast.reduce((total, cast) =>
+      total + Object.values(cast).reduce((totalPerSource, source) =>
+        totalPerSource + source, 0)
+      , 0) / this._petsPerCast.length) || 0;
+    const mergedPets = {};
+    this._petsPerCast.forEach(cast => {
+      Object.keys(cast).forEach(demonSource => {
+        mergedPets[demonSource] = (mergedPets[demonSource] || 0) + cast[demonSource];
+      });
+    });
+
+    const petTableRows = [];
+    Object.keys(mergedPets).forEach(demonSource => {
+      petTableRows.push(
+        <tr key={demonSource}>
+          <td><SpellIcon id={parseInt(demonSource)} /></td>
```

Comment:
Since there's quite a lot of space in the column, you might as well make it a SpellLink, that would clear up the confusion between Summon Felguard and Grimoire: Felguard too.
```suggestion
          <td><SpellLink id={Number(demonSource)} /></td>
```

Reply:
Fixed.

![image](https://user-images.githubusercontent.com/6363390/51003456-e22a8280-1504-11e9-9cea-10110429dcfa.png)